### ISSUE TITLE
Build error "type-length limit" with Rust 1.46 and nightly

### ISSUE THREAD

**Issue by lifan0913** [LemmyNet/lemmy#1114]
I try to run lemmy locally on my linux box, while I went to server directory and run `cargo run`, got the following issue:

error: reached the type-length limit while instantiating `<futures::future::Ready<std::res...y>::new_service::{{closure}}#0]>`
   --> /home/lifan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/futures-util-0.3.5/src/future/future/mod.rs:142:5
    |
142 | /     fn map<U, F>(self, f: F) -> Map<Self, F>
143 | |     where
144 | |         F: FnOnce(Self::Output) -> U,
145 | |         Self: Sized,
146 | |     {
147 | |         assert_future::<U, _>(Map::new(self, f))
148 | |     }
    | |_____^
    |
    = note: consider adding a `#![type_length_limit="1471465"]` attribute to your crate

error: aborting due to previous error

error: could not compile `lemmy_server`.

To learn more, run the command again with --verbose.

OS: Debian 10 LInux kernel 4.19.0-6-amd64
rust toolchain: recent nightly

**Comment by Nutomic** [comment]
Lets keep this open for now cause many people will probably run into it.

**Comment by startion2007** [comment]
Add the "rust-toolchain" file to the directory with the version number -> 1.42.0

**Comment by dessalines** [comment]
Just for reference, we're using `rustup install nightly-2020-05-07` until the rust devs fix this issue.

**Comment by Nutomic** [comment]
After updating to [Rust 1.47](https://blog.rust-lang.org/2020/10/08/Rust-1.47.html) and latest nightly, I can now build Lemmy without any errors. A bit strange that this isnt mentioned in the changelog.

Anyway if others can confirm, then we can close this issue.

**Comment by dessalines** [comment]
Mine builds, but on running with the new nightly in docker, I get this:

```
lemmy_1     | Panic in Arbiter thread.                                                                            
lemmy_1     | thread 'main' panicked at 'attempted to leave type `linked_hash_map::Node<std::string::String, value
::Value>` uninitialized, which is invalid', /rustc/91a79fb29ac78d057d04dbe86be13d5dcc64309a/library/core/src/mem/m
od.rs:659:9                                                                                                       
lemmy_1     | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace                       
```

**Comment by dessalines** [comment]
This error is for `config-rs` not working with the newest rust: https://github.com/mehcode/config-rs/issues/158

**Comment by Nutomic** [comment]
The original issue is fixed in Rust 1.47.0, and I opened #1270 for the problem with config-rs.
